### DISCUSSION TITLE
FallbackFocus executes several times. It it expected behavior?

### DISCUSSION THREAD

**Discussion by minaev007** [focus-trap/focus-trap#417] [Q&A] [answered]
When container has `display: none` **_**fallbackFocus**_** option executes, what is expected behavior, but it fires 2-3 times. Was it done deliberately?

**Comment by stefcameron** [comment, marked as answer]
Hi! 👋 So if the container you give to `createFocusTrap()` is hidden, the trap won't find any tabbable nodes inside of it, and `fallbackFocus` will be called. We also call the `fallbackFocus` function (if you provide one) at different times. First for setting the initial focus, and then a second time when we update the list of tabbable nodes when the trap is first activated. So that would be at least 2 times.

**Reply by minaev007** [reply]
Thanks for your fast and detailed answer! 💪
